components/TextFields/src/MDCTextField.m

```diff
@@ -106,6 +118,78 @@ - (void)commonMDCTextFieldInitialization {
                       object:self];
 }
 
+#pragma mark - Underline View Implementation
+
+- (void)setupUnderlineConstraints {
+  NSLayoutConstraint *underlineLeading = [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeLeading
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeLeading
+                              multiplier:1
+                                                                       constant:0];
+  underlineLeading.priority = UILayoutPriorityDefaultLow;
+  underlineLeading.active = YES;
+
+  NSLayoutConstraint *underlineTrailing = [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeTrailing
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeTrailing
+                              multiplier:1
+                                                                        constant:0];
+  underlineTrailing.priority = UILayoutPriorityDefaultLow;
+  underlineTrailing.active = YES;
+
+  _underlineY =
+  [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeCenterY
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeTop
+                              multiplier:1
+                                constant:[self textInsets].top + [self estimatedTextHeight] +
+   MDCTextInputHalfPadding];
+  _underlineY.priority = UILayoutPriorityDefaultLow;
+  _underlineY.active = YES;
+}
+
+- (CGFloat)underlineYConstant {
+  return [self textInsets].top + [self estimatedTextHeight] + MDCTextInputHalfPadding;
+}
+
+- (BOOL)needsUpdateUnderlinePosition {
+  NSInteger currentConstant = (NSInteger)(self.underlineY.constant * 1000);
+  NSInteger idealConstant = (NSInteger)([self underlineYConstant] * 1000);
+
+  return currentConstant != idealConstant;
+}
```

Comment:
Yeah this one. @romoore correctly pointed out that I was doing floating point comparison which is bad. So I need to convert these. I chose 1000 to be conservative. I believe the most decimal places It'll ever be is actually 1 which would be 10. Should I change to 10?